dlp/snippets/deid/deid_with_exception_list.go

```diff
@@ -0,0 +1,120 @@
+
+	dlp "cloud.google.com/go/dlp/apiv2"
+	"cloud.google.com/go/dlp/apiv2/dlppb"
+)
+
+// deidentifyExceptionList creates an exception list for a regular custom dictionary detector.
+func deidentifyExceptionList(w io.Writer, projectID, input, infoType string, dictWordList []string) error {
+	// projectID := "my-project-id"
+	// input := "jack@example.org accessed customer record of user5@example.com"
+	// infoType := "EMAIL_ADDRESS"
+	// dictWordList := []string{"jack@example.org", "jill@example.org"}
+
+	ctx := context.Background()
+
+	// Initialize a client once and reuse it to send multiple requests. Clients
+	// are safe to use across goroutines. When the client is no longer needed,
+	// call the Close method to cleanup its resources.
+	client, err := dlp.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("dlp.NewClient: %v", err)
+	}
+
+	// Closing the client safely cleans up background resources.
+	defer client.Close()
+
+	// Specify what content you want the service to DeIdentify.
+	var item = &dlppb.ContentItem{
+		DataItem: &dlppb.ContentItem_Value{
+			Value: input,
+		},
+	}
+
+	// Specify the word list custom info type and build-in info type the inspection will look for.
+	var infoTypes = []*dlppb.InfoType{
+		{Name: infoType},
+	}
+
+	inspectRuleSet := &dlppb.InspectionRuleSet{
+		InfoTypes: infoTypes,
+		Rules: []*dlppb.InspectionRule{
+			{
+				Type: &dlppb.InspectionRule_ExclusionRule{
+					ExclusionRule: &dlppb.ExclusionRule{
+						MatchingType: dlppb.MatchingType_MATCHING_TYPE_FULL_MATCH,
+						Type: &dlppb.ExclusionRule_Dictionary{
+							Dictionary: &dlppb.CustomInfoType_Dictionary{
+								Source: &dlppb.CustomInfoType_Dictionary_WordList_{
+									WordList: &dlppb.CustomInfoType_Dictionary_WordList{
+										Words: dictWordList,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// Construct the configuration for the de-id request and list all desired transformations.
+	var deIdentifyConfig = &dlppb.DeidentifyConfig{
+		Transformation: &dlppb.DeidentifyConfig_InfoTypeTransformations{
+			InfoTypeTransformations: &dlppb.InfoTypeTransformations{
+				Transformations: []*dlppb.InfoTypeTransformations_InfoTypeTransformation{
+					{
+						PrimitiveTransformation: &dlppb.PrimitiveTransformation{
+							Transformation: &dlppb.PrimitiveTransformation_ReplaceWithInfoTypeConfig{},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// Create a configured request.
+	req := &dlppb.DeidentifyContentRequest{
+		Parent:           fmt.Sprintf("projects/%s/locations/global", projectID),
+		DeidentifyConfig: deIdentifyConfig,
+		InspectConfig: &dlppb.InspectConfig{
+			InfoTypes:       infoTypes,
+			CustomInfoTypes: []*dlppb.CustomInfoType{},
+			RuleSet:         []*dlppb.InspectionRuleSet{inspectRuleSet},
+		},
+		// The item to analyze.
+		Item: item,
+	}
+
+	// Send the request.
+	resp, err := client.DeidentifyContent(ctx, req)
+	if err != nil {
+		return fmt.Errorf("DeidentifyContent: %v", err)
```

Comment:
issue: return the error directly to the caller.

Reply:
fixed it.